engine/crates/composition/tests/composition_tests.rs

```diff
@@ -67,4 +67,36 @@ fn run_test(federated_graph_path: &Path) -> datatest_stable::Result<()> {
     .into())
 }
 
-datatest_stable::harness! { run_test, "./tests/composition", r"^.*federated.graphql$" }
+fn test_sdl_roundtrip(federated_graph_path: &Path) -> datatest_stable::Result<()> {
```

Comment:
I opted for this instead of tests for from_sdl() specifically. It should be a pretty good way to test that any FederatedGraph we can compose matches 1 to 1 to a join spec SDL schema.